ui/apps/platform/src/utils/searchUtils.ts

```diff
@@ -35,7 +49,10 @@ export function hasSearchModifier(searchOptions, modifier) {
     );
 }
 
-export function getViewStateFromSearch(search, key) {
+export function getViewStateFromSearch(
+    search: Record<string, string | boolean>,
```

Comment:
I'm _pretty sure_ the type of `search` here is only `Record<string, string>` based on this value eventually being derived from a `qs.parse` call, and the fact that almost all other search parameters in this file are `Record<string, string>`.

However there is an explicit check against a boolean in the function body, and we have tests that check booleans, so I left it as-is to be safe.

Reply:
> ... there is an explicit check against a boolean in the function body, and we have tests that check booleans, so I left it as-is to be safe

I think the standards bodies allow query strings of the form `?loggedin&verified`, where the presence of a key in a query string is short-hand for a "true" boolean value.

In some languages, this would parse to a boolean `true`. I'm not sure if the native JS `qs` package parses it that way, but it's worth verifying in the near future.

Reply:
Oh good call. I'm _pretty sure_ that `qs` parses everything into a string, array of strings, or object with string key/vals. I was reading a long issue chain on their repo where they strongly stated that the lib would never automatically parse values into `number`, `boolean`, etc. Honestly I think `qs` would parse a lone key into `<key> = null`.

I'll verify this though.